src/System.IO.FileSystem/src/System/IO/Win32FileSystem.cs

```diff
@@ -559,6 +562,10 @@ private static void RemoveDirectoryInternal(string fullPath, bool topLevel)
                         if (!topLevel)
                             return;
                         break;
+                    case Interop.Errors.ERROR_DIR_NOT_EMPTY:
+                        if (allowDirectoryNotEmpty)
```

Comment:
Is the folder going to be removed anyway? In case of error it seems to indicate that the folder is not going to be marked for later deletion.

Reply:
It won't be removed. There really isn't a practical way around this. The argument is that it is better to dangle empty directories then throw for something you can't reliably handle.

I've tried just opening a handle with the delete flag set on a directory with a locked file in it and it just silently succeeds, but doesn't actually do anything. The file system (presumably NTFS itself) just doesn't let you mark directory entries for deletion when they contain pending deleted files.

I suppose we could add some limited retry logic in this specific case. Neither solution is great, but this is definitely a recurring gotcha for folks.